lib/commands/logs.ts

```diff
@@ -75,6 +78,11 @@ export default class LogsCmd extends Command {
 	public static usage = 'logs <device>';
 
 	public static flags: flags.Input<FlagsDef> = {
+		'max-retry': flags.integer({
+			description: stripIndent`
+				Maximum number of reconnection attempts on "connection lost" errors
+				(use 0 to disable auto reconnection).`,
```

Comment:
Sample command output, simulating connection lost with the `tcpkill` utility on the device:
```
$ ./bin/balena logs 192.168.1.46 --max-retry 2
[Logs]    [12/8/2020, 12:37:11 AM] [main] Plain Dockerfile 47 (11602) howdy (5)
[Logs]    [12/8/2020, 12:37:21 AM] [main] Plain Dockerfile 47 (11603) howdy (5)
[Logs]    [12/8/2020, 12:37:31 AM] [main] Plain Dockerfile 47 (11604) howdy (5)
[Warn]    Connection to device lost
Retrying "Streaming logs" after 1.0s (1 of 2) due to: DeviceConnectionLostError: Connection to device lost
[Warn]    Connection to device lost
Retrying "Streaming logs" after 2.0s (2 of 2) due to: DeviceConnectionLostError: Connection to device lost
[Logs]    [12/8/2020, 12:37:41 AM] [main] Plain Dockerfile 47 (11605) howdy (5)
[Logs]    [12/8/2020, 12:37:51 AM] [main] Plain Dockerfile 47 (11606) howdy (5)
[Warn]    Connection to device lost
Max retry count (2) exceeded while attempting to reconnect to the device
```
... and then the process exited with error code 1, as expected.